src/StreamFactoryInterface.php

```diff
@@ -9,15 +9,11 @@
     /**
      * Create a new stream from a resource or a string.
      *
-     * If the argument is a resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
-     * If the argument is a string, a temporary resource will be created that is writable and seekable.
-     * 
-     * If the arugment is a string it will be interpided as the content of the stream. File names or 
-     * file paths are not supported. 
+     * String content will be used as the body of the stream. Files paths are not supported.
```

Comment:
@weierophinney yes, but is that an interoperability problem? I don't think we're supposed to be designing a library here - we're supposed to think less about utility, and more about reducing things to the level of being interoperable.

If we can reduce interoperability concerns to a single, consistent interface, and this enables someone to write a utility library that provides the same functionality and utility for *all* implementations of that interface, isn't that the ideal?

In my opinion, if we design the interfaces not based on what's minimally required to level the differences between libraries, but instead design it to meet our own personal ideals or preferences about what such a library should look like, we're unnecessarily burdening developers with extra work to satisfy our needs.

I think it's really important to remember that we're not designing a library here - we're designing the layer that is supposed to *help* libraries be interchangible.

If resource-based streams are the only use-case we're going to cover in terms of interoperability, the only interface we strictly must have, is:

    public function createStream($resource);

This covers all use-cases *indirectly* - as demonstrated above, with this simple factory interface implemented, we can easily provide a single library (and others can provide competing ones, if they feel so inclined) to create and populate temp-streams, open files, etc. without burdening everyone to implement those again and again, and without actually designing the library/API *for* the implementers. I don't think that's our place.